warehouse/query-core/src/main/java/datawave/query/jexl/functions/TLDEventFieldAggregator.java

```diff
@@ -0,0 +1,34 @@
+package datawave.query.jexl.functions;
+
+import datawave.query.predicate.EventDataQueryFilter;
+import datawave.query.tld.TLD;
+import datawave.query.util.TypeMetadata;
+import org.apache.accumulo.core.data.ByteSequence;
+
+import java.util.ArrayList;
+
+public class TLDEventFieldAggregator extends EventFieldAggregator {
+    public TLDEventFieldAggregator(String field, EventDataQueryFilter filter, int maxNextCount, TypeMetadata typeMetadata, String defaultTypeClass) {
+        super(field, filter, maxNextCount, typeMetadata, defaultTypeClass);
+    }
+    
+    @Override
+    protected ByteSequence parsePointer(ByteSequence columnFamily) {
+        // find the null between the dataType and Uid
+        ArrayList<Integer> nulls = TLD.instancesOf(0, columnFamily, 1);
+        final int start = nulls.get(0) + 1;
+        
+        // uid is from the null byte to the end of the cf
+        ByteSequence uid = columnFamily.subSequence(start, columnFamily.length());
+        
+        // find the end of the tld if it exists
+        ArrayList<Integer> dots = TLD.instancesOf('.', uid);
+        if (dots.size() > 2) {
+            // reduce to the TLD
+            return columnFamily.subSequence(0, start + dots.get(2));
```

Comment:
I think this is making an assumption about how we create UIDs.  If we turned on the timestamp portion of the UID then this would break or if we decide to substitute an alternate UID mechanism as it is currently pluggable.  Can we simply look for the last dot and strip that off?

Reply:
What if there's more than one dot.

Reply:
I updated the code to use UID to get the base UID

Reply:
Generally this query logic is probably not configured right if the UID scheme changes